### PR TITLE
Adding asset configuration parameter - option to save raw email to container data dictionary

### PR DESCRIPTION
**Pull request by tonyc-phantom**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by tonyc-phantom** [comment]
@sodle-splunk @alexa-phantom @splunk-jessica I tested upgrading from the previous version of the app to this one, and it worked but will you please verify my logic? The previous version did not have the check box so I want to make sure it will continue to work without having to edit the asset first. That is why I chose the default setting to be true.

File: process_email.py [hunk 1]
```diff
@@ -959,7 +959,11 @@ def _handle_mail_object(self, mail, email_id, rfc822_email, tmp_dir, start_time_
         del (container_data[PROC_EMAIL_JSON_EMAIL_HEADERS])
         self._container['source_data_identifier'] = email_id
         self._container['name'] = container_name
-        self._container['data'] = {'raw_email': rfc822_email}
+
+        # Save raw email to data dictionary in container
+        if self._config.get('save_raw_email_to_container', True):
+            self._container['data'] = {'raw_email': rfc822_email}
```
Comment (sodle-splunk) [inline process_email.py:965]: Is it safe for the container to not have a `data` property? Did that work ok when testing?

Wondering if it's better to leave it unset, or to set it to a None or an empty object.

Comment (splunk-jessica) [reply process_email.py:965]: I think the value of `rfc822_email` is set to `None` in the `_get_email_data_from_container` function already  https://github.com/splunk-soar-connectors/microsoftexchangeonpremews/blob/e71b62204d29faa2c834a7d8cf5fc839b02ca21e/ewsonprem_connector.py#L743

**Review by splunk-jessica — APPROVED** [review]

File: process_email.py (hunk 1 above)
Comment (tonyc-phantom) [reply process_email.py:965]: @sodle-splunk The data dictionary is there by default, and with my current logic I am not adding the 'raw_email' key if the box is unchecked. The end result is that the data dict is empty, the `raw_email` key does not exist, and that action that @splunk-jessica mentioned above will fail as I would expect it to. My thinking was that if I just added the key anyway, users would get the wrong message like the email was there but blank, there was an error, etc.

Comment (tonyc-phantom) [reply process_email.py:965]: By the way I have tested different scenarios with the current logic and they have passed. I am submitting MRs shortly.

Comment (sodle-splunk) [reply process_email.py:965]: If it works ok with the current logic, I'd say it's probably fine...

**Review by sodle-splunk — APPROVED** [review]

**Pull request merged**